src/m365/entra/commands/m365group/m365group-set.ts

```diff
@@ -149,25 +179,27 @@ class EntraM365GroupSetCommand extends GraphCommand {
     await this.showDeprecationWarning(logger, aadCommands.M365GROUP_SET, commands.M365GROUP_SET);
 
     try {
+      if ((args.options.allowExternalSenders !== undefined || args.options.autoSubscribeNewMembers !== undefined) && accessToken.isAppOnlyAccessToken(auth.connection.accessTokens[auth.defaultResource].accessToken)) {
+        throw `Option 'allowExternalSenders' and 'autoSubscribeNewMembers' can only be used when using delegated permissions.`;
+      }
+
       const isUnifiedGroup = await entraGroup.isUnifiedGroup(args.options.id);
 
       if (!isUnifiedGroup) {
         throw Error(`Specified group with id '${args.options.id}' is not a Microsoft 365 group.`);
       }
 
-      if (args.options.displayName || args.options.description || typeof args.options.isPrivate !== 'undefined') {
-        if (this.verbose) {
-          await logger.logToStderr(`Updating Microsoft 365 Group ${args.options.id}...`);
-        }
+      if (this.verbose) {
+        await logger.logToStderr(`Updating Microsoft 365 Group ${args.options.id}...`);
+      }
 
+      if (args.options.displayName || args.options.description || args.options.isPrivate !== undefined) {
         const update: Group = {};
-        if (args.options.displayName) {
-          update.displayName = args.options.displayName;
-        }
-        if (args.options.description) {
-          update.description = args.options.description;
-        }
-        if (typeof args.options.isPrivate !== 'undefined') {
+
+        update.displayName = args.options.displayName;
```

Comment:
Could we add a small enhancement to clear this value? If the user passes an empty string `""`, let's set the value to `null` to clear it.